testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCTestCase.java

```diff
@@ -0,0 +1,227 @@
+    private static final String MODULE_NAME = "clustered2lc";
+
+    @ArquillianResource
+    protected ContainerController controller;
+
+    @ArquillianResource
+    protected Deployer deployer;
+
+    @Deployment(name = DEPLOYMENT_1, managed = false, testable = false)
+    @TargetsContainer(CONTAINER_1)
+    public static Archive<?> createDeploymentForContainer1() {
+        return createDeployment();
+    }
+
+    @Deployment(name = DEPLOYMENT_2, managed = false, testable = false)
+    @TargetsContainer(CONTAINER_2)
+    public static Archive<?> createDeploymentForContainer2() {
+        return createDeployment();
+    }
+
+    private static Archive<?> createDeployment() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, MODULE_NAME + ".war");
+        war.addPackage(ClusteredJPA2LCTestCase.class.getPackage());
+        war.addAsWebInfResource(ClusteredJPA2LCTestCase.class.getPackage(), "persistence.xml",
+                "classes/META-INF/persistence.xml");
+        log.info(war.toString(true));
+        return war;
+    }
+
+    // management connection to node0
+    private ModelControllerClient client0;
+
+    // management connection to node1
+    private ModelControllerClient client1;
+
+    // REST client to control entity creation, caching, eviction,... on the servers
+    private Client restClient = ClientBuilder.newClient();
+
+    // /subsystem=infinispan/cache-container=hibernate/replicated-cache=entity-replicated
+    static ModelNode CACHE_ADDRESS;
+
+    static {
+        CACHE_ADDRESS = new ModelNode();
+        CACHE_ADDRESS.get("subsystem").set("infinispan");
+        CACHE_ADDRESS.get("cache-container").set("hibernate");
+        CACHE_ADDRESS.get("replicated-cache").set("entity-replicated");
+    }
+
+    @Test
+    @InSequence(-1)
+    public void setupCacheContainer() throws IOException {
+        NodeUtil.start(controller, CONTAINERS);
+
+        final ModelNode createEntityReplicatedCacheOp = new ModelNode();
+        createEntityReplicatedCacheOp.get(ADDRESS).set(CACHE_ADDRESS);
+        createEntityReplicatedCacheOp.get(OP).set(ADD);
+        createEntityReplicatedCacheOp.get("mode").set("sync");
+
+        client0 = createClient0();
+        client1 = createClient1();
+
+        final ModelNode result0 = client0.execute(createEntityReplicatedCacheOp);
+        Assert.assertTrue(result0.toJSONString(false), result0.get(OUTCOME).asString().equals(SUCCESS));
+
+        final ModelNode result1 = client1.execute(createEntityReplicatedCacheOp);
+        Assert.assertTrue(result1.toJSONString(false), result1.get(OUTCOME).asString().equals(SUCCESS));
+
+        NodeUtil.deploy(this.deployer, DEPLOYMENTS);
+    }
+
+    /**
+     * We have a replicated entity cache between two nodes. Cache an entity on node0 and then verify that
+     * node1 sees the cached entity as well. Then try evicting it on one node and see if it's been evicted
```

Comment:
Yeah I can add that. Actually that sounds like an idea for another test to add - where eviction would be done with TransactionAttributeType.NEVER. Does that make sense? Not sure how safe that is though, can't it cause intermittent failures due to race conditions or something?